src/tags/tests/test_models.py

```diff
@@ -0,0 +1,204 @@
+from django_tenants.test.cases import TenantTestCase
+from django.contrib.auth import get_user_model
+
+from model_bakery import baker
+
+from tags.models import total_xp_by_tags
+from siteconfig.models import SiteConfig
+from quest_manager.models import Quest, QuestSubmission
+from badges.models import Badge, BadgeAssertion
+
+import random
+import itertools
+
+User = get_user_model()
+
+
+class Tag_total_xp_by_tags_Tests(TenantTestCase):
+    """ 
+        Specialized TestClass for testing total_xp_by_tags function
+    """ 
+
+    def setUp(self):
+        self.user = baker.make(User)
+        self.random = random.Random(9921622)
+
+    def create_quest(self, xp):
+        quest = baker.make(Quest, xp=xp)
+        qs = baker.make(
+            QuestSubmission, 
+            quest=quest, 
+            user=self.user, 
+            is_completed=True, 
+            is_approved=True, 
+            semester=SiteConfig().get().active_semester,
+        )
+
+        return quest, qs
+
+    def create_badge(self, xp):
+        badge = baker.make(Badge, xp=xp)
+        ba = baker.make(BadgeAssertion, badge=badge, user=self.user)
+
+        return badge, ba
+
+    # QUEST ONLY TESTS
+
+    def test_one_tag_quests_only(self):
+        """ 
+            See if correct xp is displayed using only 1 tag added to Quest objects
+        """ 
+        xp_list = [self.random.randint(0, 100) for i in range(10)]
```

Comment:
don't use random numbers in tests.  Tests need to be deterministic, running the identically each time, by introducing an element that can't be repeated, it reduces the value of the test.  Pick a reasonable number and use that.  I know it seems to make sense to pick a random number, to cover more possible cases, however when you run a test you are only using ONE of those possibility, but when it fails,  you don't know which one, which can make the test useless.  It would make more sense to run the test 100 times, once for each value, but then our testing suite would take forever.   So, just pick a reasonable (smallish for speed) number of items to create.

Reply:
I should add, I used to do this too (it SEEMS logical to do!) but have learned since then, haha.  There's lots online about this, e.g. https://softwareengineering.stackexchange.com/a/221636

Reply:
im using a seed for random
![image](https://user-images.githubusercontent.com/39788517/181388963-502a5103-5cd5-424e-8b76-ed649c61b426.png)
so it should be the same set of numbers everytime
```[55, 64, 1, 34, 38, 42, 90, 24, 46, 100]``` are the first 10 set of numbers i get when i use ```self.random.randint(0, 100)```

Reply:
So this seems to make the use of random redundant?  All these tests will be using 55 if that's the case.  So why not just use 55 explicitly?  Also, because we run tests multithreaded/parallel, there's no guarantee even that reseeding will give the same result (at least according to the docs for Random)

I'm going to pull out the Zen of Python:
`Explicit is better than implicit`
`Simple is better than complex`

So, remove the random code, and just provide that list explicitly `xp_list = [55, 64, 1, 34, 38, 42, 90, 24, 46, 100]`

I do like the idea of randomly generating that list in the first place though, but now that we have it, lets make sure these tests are deterministic, explicit, and a little more simple (and easier to read) by removing the random code.

Reply:
Fixed
Added a list of 50 ints that im calling/slicing instead of calling self.random.randint()